executor/merge_join.go

```diff
@@ -109,16 +111,31 @@ func (t *mergeJoinInnerTable) init(ctx context.Context, chk4Reader *chunk.Chunk)
 	return err
 }
 
-func (t *mergeJoinInnerTable) rowsWithSameKey() ([]chunk.Row, error) {
+func (t *mergeJoinInnerTable) rowsWithSameKey(ctx sessionctx.Context) (*chunk.RowContainer, error) {
 	lastResultIdx := len(t.resultQueue) - 1
 	t.resourceQueue = append(t.resourceQueue, t.resultQueue[0:lastResultIdx]...)
 	t.resultQueue = t.resultQueue[lastResultIdx:]
+	if t.sameKeyRows == nil {
+		rc := chunk.NewRowContainer(t.reader.base().retFieldTypes, t.curResult.Capacity())
```

Comment:
Would it be better that `chunk.NewRowContainer` returns a pointer of `RowContainer`?

Reply:
Yes, will do.